packages/env/lib/config/parse-config.js

```diff
@@ -300,12 +297,20 @@ async function parseRootConfig( configFile, rawConfig, options ) {
 		checkPort( configFile, `testsPort`, rawConfig.testsPort );
 		parsedConfig.testsPort = rawConfig.testsPort;
 	}
-	if ( rawConfig.afterSetup !== undefined ) {
-		// Support null as a valid input.
-		if ( rawConfig.afterSetup !== null ) {
-			checkString( configFile, 'afterSetup', rawConfig.afterSetup );
+	if ( rawConfig.lifecycleScripts ) {
+		parsedConfig.lifecycleScripts = {};
+
+		for ( const key in rawConfig.lifecycleScripts ) {
+			if ( rawConfig.lifecycleScripts[ key ] !== null ) {
+				checkString(
+					configFile,
+					key,
+					rawConfig.lifecycleScripts[ key ]
+				);
+			}
+			parsedConfig.lifecycleScripts[ key ] =
```

Comment:
Similar to the map suggestion above, it'd be nice to not allow random input here. (Maybe also solved by https://github.com/WordPress/gutenberg/pull/50642?)

Reply:
#50642 _would_ solve this, but, I need to make a change to support the requirements of this specific option. That's actually why this PR is failing right now too.